<launch>
    <arg name="robot_name" default="musc_shoulder"/>
    <arg name="robot_model" default="oxford"/>
    <arg name="model_name" default="$(arg robot_name)"/>
    <arg name="urdf_file_name" default="model.urdf"/>
    <!--<arg name="cardsflow_file_name" default="cardsflow.xml"/>-->
    <!--<arg name="start_controllers" default="head_axis0 head_axis1 head_axis2"/>-->
     <!-- <arg name="start_controllers" default="$(arg robot_model)_head_axis0 $(arg robot_model)_head_axis1 $(arg robot_model)_head_axis2 $(arg robot_model)_shoulder_left_axis0 $(arg robot_model)_shoulder_left_axis1 $(arg robot_model)_shoulder_left_axis2 $(arg robot_model)_elbow_left_axis0 $(arg robot_model)_elbow_left_axis1"/> -->
    <arg name="start_controllers" default="$(arg robot_model)_$(arg robot_name)_axis0 $(arg robot_model)_$(arg robot_name)_axis1 $(arg robot_model)_$(arg robot_name)_axis2"/>
    <arg name="simulated" default="false"/>
    <arg name="gazebo" default="false"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="external_robot_state" default="false"/>
    <arg name="external_robot_target" default="false"/>
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>
    <arg name="physics" default="ode"/>

    <param name="robot_name" type="string" value="$(arg robot_name)"  />
    <param name="model_name" type="string" value="$(arg model_name)"  />
    <param name="robot_description" textfile="$(find robots)/$(arg model_name)/model.urdf" />
    <param name="urdf_file_path" type="string" value="$(find robots)/$(arg robot_name)/$(arg urdf_file_name)" />
    <param name="cardsflow_xml" type="string" value="$(find robots)/$(arg model_name)/cardsflow.xml" />
    <!--<param name="cardsflow_xml" type="string" value="$(find robots)/$(arg model_name)/$(arg cardsflow_file_name" />-->
    <param name="simulated" type="bool" value="$(arg simulated)"/>
    <rosparam file="$(find robots)/$(arg model_name)/config/controller.yaml" command="load"/>
    <rosparam file="$(find robots)/$(arg model_name)/config/endeffectors.yaml" command="load"/>
    <rosparam file="$(find robots)/$(arg model_name)/config/motor_config.yaml" command="load"/>
    <rosparam file="$(find robots)/$(arg model_name)/config/link_joint_relation.yaml" command="load"/>
    <rosparam file="$(find robots)/$(arg model_name)/config/control_parameters.yaml" command="load"/>
    <rosparam param="q_target">[0.0, 0.0, 0.0]</rosparam>
    <rosparam param="pwm">3.0</rosparam>
    <rosparam param="m3_pwm">5.0</rosparam>
    <param name="min_force" type="double" value="0"  />
    <param name="max_force" type="double" value="100000"  />
    <param name="controller" type="int" value="2"  />
    <param name="external_robot_state" type="bool" value="$(arg external_robot_state)"/>

    <node name="$(arg robot_model)_controller_manager"
          pkg="controller_manager"
          type="spawner"
          args="$(arg start_controllers)"

/>

    <node name="$(arg robot_model)_robot_controller"
         pkg="kindyn"
         type="$(arg robot_name)"
         args="$(arg robot_model) $(arg debug)"
         respawn="false"
         output="screen"
   />


    <group if="$(arg gazebo)">
        <include file="$(find gazebo_ros)/launch/empty_world.launch">
            <arg name="debug" value="$(arg debug)"/>
            <arg name="gui" value="$(arg gui)"/>
            <arg name="paused" value="$(arg paused)"/>
            <arg name="use_sim_time" value="$(arg use_sim_time)"/>
            <arg name="headless" value="$(arg headless)"/>
            <arg name="physics" value="$(arg physics)"/>
        </include>
        <node name="robot_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
              args="-sdf -file $(find robots)/$(arg model_name)/model.sdf -model $(arg model_name)
              -x $(arg x) -y $(arg y) -z $(arg z)"
        />
    </group>


</launch>
        <!-- launch-prefix="gdb -ex run - - args"-->
